java/Anton-Kozyk/src/com/zoolatech/lecture6/tasks/_2/Order.java

```diff
@@ -0,0 +1,33 @@
+package com.zoolatech.lecture6.tasks._2;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Order {
+    private String id;
+    private String country;
+    private float price;
+    private Marketplace marketplace;
+
+    public float getPrice() {
+        return price;
+    }
+
+    public Order(String id, String country, float price, String marketplace) {
+        this.id = id;
+        this.country = country;
+        this.price = price;
+        this.marketplace = Marketplace.valueOf(marketplace.toUpperCase());
+    }
+
+    public static Map<String, Double> averagePrice(List<Order> orders) {
+        Map<String, Order> uniqOrders = orders.stream() //with this map we can be sure that no two orders will have the same id
```

Comment:
nice, but could you please find a way how to achieve the same without a second map?